### ISSUE TITLE
model_stage3代码问题

### ISSUE THREAD

**Issue by itis112** [yangmengmeng1997/APFNet#15]
output = torch.bmm(affinity, w_v) 
        # output=output.permute(0,2,1)
        # output=self.transformer1_FFN[1](nn.Dropout(0.2)(F.relu(((self.transformer1_FFN[0](output))))))
        # output=output.permute(0,2,1)
        output=output.reshape(batch,dim,w,h)
        output=self.transformer1_encoder1[3](output)  
        x=x+output   
        return x
你好，我想问一下按照您论文里面figure3的encoder的部分是先进行了自注意力后再add再升通道的，但是按照您model_stage3.py里面就是上面的代码是进行了自注意力后先升通道再与原始值相加的，请问是我理解有问题吗？

**Comment by yangmengmeng1997** [comment]
设计线性降维是为了减少计算注意力机制时内存占用过大的问题，后面需要升维来还原原来的维度，这样才可以和原始的tensor相加，按照代码理解即可，
不好意思图示给您造成误导了。




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "yangmengmeng1997/APFNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年9月27日(星期三) 下午4:16
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[yangmengmeng1997/APFNet] model_stage3代码问题 (Issue #15)





     output = torch.bmm(affinity, w_v)      # output=output.permute(0,2,1)     # output=self.transformer1_FFN[1](nn.Dropout(0.2)(F.relu(((self.transformer1_FFN[0](output))))))     # output=output.permute(0,2,1)     output=output.reshape(batch,dim,w,h)     output=self.transformer1_encoder1[3](output)       x=x+output        return x  
你好，我想问一下按照您论文里面figure3的encoder的部分是先进行了自注意力后再add再升通道的，但是按照您model_stage3.py里面就是上面的代码是进行了自注意力后先升通道再与原始值相加的，请问是我理解有问题吗？
